File: python/bsd_insert_after.py
import random
import pprint

class node:
	def __init__(self, nxt = None, last = None):
		self.next = nxt
		self.last = last

#############
# acc func  #
#############
def create_queue(size):
	hd = node()
	hd.next = None
	hd.last = hd
	for i in range(0, size):
		new_node = node()
		new_node.next = None
		lst = hd.last
		lst.next = new_node
		hd.last = new_node
#############
# end acc   #
#############

def insert_after(head, listelm):
	elm = node()
	lenn = listelm.next
	elm.next = lenn
	if lenn is None:
		head.last = elm
	listelm.next = elm
	return head

def main():
	hd = create_queue(5)
	elm = hd.next
	insert_after(hd, elm)

if __name__=="__main__":
	main()

File: benchmarks/java/retrieve heap/src/testsuites/BSD_Insert_After.java
package testsuites;

public class BSD_Insert_After {

    static Node create_queue(int size){
        Node hd = new Node();
        hd.next = null;
        hd.last = hd;

        for(int i = 0; i < size; i++){
            Node new_node = new Node();
            new_node.next = null;
            Node  lst = hd.last;
            lst.next = new_node;
            hd.last = new_node;
        }
        return hd;
    }

    static Node simpleq_insert_after(Node  head, Node listelm)
    {
        //pre
        Node  elm = new Node();
        Node  lenn = listelm.next;
        elm.next = lenn;
        if (lenn == null) {
            head.last = elm;
        }
        listelm.next = elm;
        //post
        return head;
    }

    public static void main(String args[]){
        Node hd = create_queue(10);
        Node elm = hd.next;
        simpleq_insert_after(hd, elm);
    }
}

class Node{
    Node next;
    Node last;
}
